lib/pbench/server/database/models/metadata.py

```diff
@@ -0,0 +1,73 @@
+import datetime
+from pbench.server.database.database import Database
+from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text
+
+
+class UserMetadata(Database.Base):
+    """ Metadata Model for storing user metadata details """
+
+    # TODO: Think about the better name
```

Comment:
Hey @npalaska, this might be a silly question but what do we mean by user metadata here, in context with the dashboard? I believe that the user (via dashboard) has to send some data to the REST API but do not know what it is

Reply:
@aquibbaig, I believe the intention is to provide Server-side storage for any user state which should persist between login sessions, anything which the Dashboard might want to store on behalf of the user, such as favorite datasets or dataset tags or labels, views, window geometries, default languages, or other settings, etc., any context which is associated with a user rather than a dataset.  

Storing user-generated favorites/tags/labels with datasets doesn't work well if the user doesn't own the dataset or if different users want to mark the same dataset differently.  Users might want to "favorite" a dataset published by someone else.  Since the notion of "favorite datasets" directs what the user sees, it's an attribute of the user, not the dataset.  The metadata gives the Dashboard a place to store this stuff so that it can be fetched each time the user logs in.

The "metadata" API is supposed to be very generic, to give it maximum flexibility for supporting the Dashboard and any other client uses that we haven't imagined yet.